iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -137,99 +154,440 @@ public static RegistryManager getRegistyManager(String iotHubConnectionString) t
+                    else
+                    {
+                        log.debug("Requeueing device {} into testSasDeviceQueue", testIdentity.getDeviceId());
+                        testSasDeviceQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                }
+                else
+                {
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing device {} into testX509DeviceWithTwinQueue", testIdentity.getDeviceId());
+                        testX509DeviceWithTwinQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing device {} into testX509DeviceQueue", testIdentity.getDeviceId());
+                        testX509DeviceQueue.add((TestDeviceIdentity) testIdentity);
+                    }
+                }
+            }
+            else if (testIdentity instanceof TestModuleIdentity)
+            {
+                if (((TestModuleIdentity) testIdentity).getModule().getAuthenticationType() == AuthenticationType.SAS)
+                {
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing module {} into testSasModuleWithTwinQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testSasModuleWithTwinQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing module {} into testSasModuleQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testSasModuleQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                }
+                else
+                {
+                    if (testIdentity.twinUpdated)
+                    {
+                        log.debug("Requeueing module {} into testX509ModuleWithTwinQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testX509ModuleWithTwinQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                    else
+                    {
+                        log.debug("Requeueing module {} into testX509ModuleQueue", ((TestModuleIdentity) testIdentity).getModuleId());
+                        testX509ModuleQueue.add((TestModuleIdentity) testIdentity);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void disposeTestDevices(Iterable<Device> devices, String iotHubConnectionString)
+    {
+        List<Device> subList = new ArrayList<>();
+        for (Device device : devices)
+        {
+            // Bulk device removal is limited to, at most, 100 devices per request. Create a list of 100 devices,
+            // delete them, and then create another list of 100 devices until all devices have been deleted
+            subList.add(device);
```

Comment:
Nit: More of a personal preference with respect to code clarity -
```
if(subList.size() <= 100)
{
   subList.add(device);
}
else
{
}
```

Reply:
I'm not sure I follow how you picture this change. This if statement is within a foreach loop, so I always want to add each device to the sublist. It's just when the sublist gets big enough that something different happens